quicklens/src/test/scala-3/com/softwaremill/quicklens/test/ExtensionCopyTest.scala

```diff
@@ -0,0 +1,89 @@
+package com.softwaremill.quicklens
+package test
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+object ExtensionCopyTest {
+  case class V(x: Double, y: Double, z: Double)
+
+  opaque type Vec = V
+
+  object Vec {
+    def apply(x: Double, y: Double): Vec = V(x, y, 0)
+
+    extension (v: Vec) {
+      def x: Double = v.x
+      def y: Double = v.y
+      def copy(x: Double = v.x, y: Double = v.y): Vec = V(x, y, 0)
+    }
+  }
+}
+
+class ExtensionCopyTest extends AnyFlatSpec with Matchers {
+  /*
```

Comment:
I assume this test doesn't pass - maybe you could add a comment with the reason?

Reply:
Comment added.